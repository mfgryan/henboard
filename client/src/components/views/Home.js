// react dep
import React, { Component } from "react";
import { connect } from "react-redux";
import projects from "../../models/projects.js";
import lanes from "../../models/lanes.js";

// component dep
import App from "../App";
import Swimlane from "../containers/Swimlane";
import SelectorBoard from "../containers/SelectorBoard";

// style dep
import { Col, Row } from "react-bootstrap";
import "../../css/Home.css";

export const Home = ({ project, lanes }) => {
    const getColWidth = (laneCount, index) =>
        12 / laneCount + (index < 12 % laneCount ? 1 : 0);
    return (
        <App>
            <div>
                <Row className="board">
                    {lanes.map((lane, index) =>
                        <Col md={getColWidth(lanes.length, index)} key={index}>
                            <Swimlane
                                project={project.project}
                                column={lane.column}
                            />
                        </Col>
                    )}
                </Row>
                <Row>
                    <Col md={12}>
                        <SelectorBoard project={project.project} />
                    </Col>
                </Row>
            </div>
        </App>
    );
};

const mapStateToProps = state => {
    let project = projects.getCurrentProject(state);
    return {
        project: project,
        lanes: lanes.getLanes(state, project.project)
    };
};

const mapDispatchToProps = dispatch => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
