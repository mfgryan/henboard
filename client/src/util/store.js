// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
