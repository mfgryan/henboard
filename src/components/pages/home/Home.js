import React, { Component } from "react";
import { Grid, Row, Col } from "react-bootstrap";
import Header from "../../header/Header";
import Board from "../../board/Board";
import Selector from "../../selector/Selector";

class Home extends Component {
    render() {
        const options = this.props.sprint.map((curSprint,index) => 
            ({val: curSprint.week, name: curSprint.week})
        );
        return (
            <Grid>
                <Header name={project.name} options={["backlog","project"]} />
                <Board lanes={project.sprint[0].lanes} />
                <Selector options={options}/>
            </Grid>
        );
    }
}
export default Home;
