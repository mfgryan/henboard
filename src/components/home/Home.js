import React, { Component } from "react";
import { Grid, Row, Col } from "react-bootstrap";
import App from "../App";
import Board from "../board/Board";
import Selector from "../selector/Selector";
import "./Home.css";

class Home extends Component {
    constructor(props){
        super(props);
    }
    loadData(){
         
    }
    render() {
        return (
            <App>
                {/*<Board lanes={data[0].sprint[0].lanes} />*/}
                {/*<Selector sprint={data[0].sprint} />*/}
            </App>
        );
    }
}
export default Home;
