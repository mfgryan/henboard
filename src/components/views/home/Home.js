import React, { Component } from "react";
import { Grid, Row, Col } from "react-bootstrap";
import App from "../../App";
import GetBoard from "../../containers/getBoard/GetBoard";
//import Selector from "../../presenters/selector/Selector";
import "./Home.css";

class Home extends Component {
    render() {
        return (
            <App>
                <GetBoard project="henboard" />
                {/*<Selector/>*/}
            </App>
        );
    }
}
export default Home;
