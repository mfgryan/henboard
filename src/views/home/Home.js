// react dep
import React, { Component } from "react";

// bootstrap dep
import { Grid, Row, Col } from "react-bootstrap";

// component dep
import App from "../../components/App";
import GetBoard from "../../components/board/GetBoard";

//style dep
import "./Home.css";

class Home extends Component {
    render() {
        return (
            <App>
                <GetBoard project="henboard" />
            </App>
        );
    }
}
export default Home;
