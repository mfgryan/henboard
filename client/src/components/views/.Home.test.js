import React from "react";
import Home from "./Home";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it('Home renders correctly', () => {
    const tree = renderer.create(
        <Router>
            <Provider store={store}>
                <Home project="henboard" lanes={store.getState().lanes} />
            </Provider>
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
