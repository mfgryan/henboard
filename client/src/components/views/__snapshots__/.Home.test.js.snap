exports[`test Home renders correctly 1`] = `
<div
  className="container">
  <div
    className="Header row">
    <div
      className="col-md-4" />
    <div
      className="HeaderDropdownCol col-md-4 col-md-offset-8">
      <div
        className="dropdown btn-group">
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="HeaderDropdownMenu dropdown-toggle btn btn-default"
          disabled={false}
          id="HeaderMenu"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button">
          ≡
        </button>
        <ul
          aria-labelledby="HeaderMenu"
          className="dropdown-menu"
          role="menu">
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/account"
              onClick={[Function]}>
              Sign In
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/home"
              onClick={[Function]}>
              home
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/backlog"
              onClick={[Function]}>
              backlog
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/planning"
              onClick={[Function]}>
              planning
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              undo 
              <small>
                Ctrl-left
              </small>
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              redo 
              <small>
                Ctrl-right
              </small>
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/settings"
              onClick={[Function]}>
              settings
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="board row">
      <div
        className="col-md-4">
        <div
          className="swimlane">
          <div
            className="AddItem">
            <h2
              onClick={[Function]}>
              Todo
            </h2>
            <div
              className="AddItemInput"
              style={
                Object {
                  "visibility": "hidden",
                }
              }>
              <div
                className="form-group">
                <span
                  className="input-group input-group-sm">
                  <input
                    className="form-control"
                    id={undefined}
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    type="text"
                    value="" />
                  <span
                    className="input-group-btn"
                    type="text">
                    <button
                      className="btn btn-default"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Add
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="itemsArea"
            onDragOver={[Function]}
            onDrop={[Function]} />
        </div>
      </div>
      <div
        className="col-md-4">
        <div
          className="swimlane">
          <div
            className="AddItem">
            <h2
              onClick={[Function]}>
              Dev
            </h2>
            <div
              className="AddItemInput"
              style={
                Object {
                  "visibility": "hidden",
                }
              }>
              <div
                className="form-group">
                <span
                  className="input-group input-group-sm">
                  <input
                    className="form-control"
                    id={undefined}
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    type="text"
                    value="" />
                  <span
                    className="input-group-btn"
                    type="text">
                    <button
                      className="btn btn-default"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Add
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="itemsArea"
            onDragOver={[Function]}
            onDrop={[Function]} />
        </div>
      </div>
      <div
        className="col-md-4">
        <div
          className="swimlane">
          <div
            className="AddItem">
            <h2
              onClick={[Function]}>
              Done
            </h2>
            <div
              className="AddItemInput"
              style={
                Object {
                  "visibility": "hidden",
                }
              }>
              <div
                className="form-group">
                <span
                  className="input-group input-group-sm">
                  <input
                    className="form-control"
                    id={undefined}
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    type="text"
                    value="" />
                  <span
                    className="input-group-btn"
                    type="text">
                    <button
                      className="btn btn-default"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Add
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="itemsArea"
            onDragOver={[Function]}
            onDrop={[Function]} />
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-md-12">
        <div
          className="dropdown btn-group">
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={false}
            id="Selector"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button">
            Sprint Week  05/15/17
          </button>
          <ul
            aria-labelledby="Selector"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                1
                  
                <small>
                  05/15/17
                </small>
              </a>
            </li>
            <li
              onKeyDown={[Function]}
              onSelect={[Function]}>
              <a
                href="#"
                onClick={[Function]}>
                <b>
                  load all
                </b>
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                New Sprint +
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12"
      id="footer">
      <i>
        Henboard 2017
      </i>
    </div>
  </div>
</div>
`;
